lib/logstash/codecs/identity_map_codec.rb

```diff
@@ -52,14 +52,15 @@ def start
         while running? do
           sleep @interval
           break if !running?
+          break if @listener.nil?
```

Comment:
this is weird, `@listener` should only be nil at the end of `stop`, and at that point `running?` will be false. any thoughts on why this is happening?

Reply:
No idea. My original understanding of the flow is as you indicated.
Ruby docs
`alive?` Returns true if thr is running or sleeping.
`wakeup` Marks a given thread as eligible for scheduling, however it may still remain blocked on I/O.

It is not logical, captain, yet it is seen in the wild and by me when running hours of repeated tests on the file input (one test uses the IMC/ML codec).

Even thinking that there might be two threads running with a reference to the same `@listener` does not make sense because they should also have a reference to the same `@running` variable - unless they don't.

The issue was first seen on Mar 1, 2016, before config reloading I think.

Since applying the fix locally the file input tests have not failed with this error after two+ hours of continual running.